client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -80,17 +89,37 @@ interface JsonFormOptions {
 }
 
 export const useJsonForms = (
-  docName: string | undefined,
+  docName: string,
   options: JsonFormOptions = {}
 ) => {
-  const [data, setData] = useState<JsonData>(patient); // Replace with DB query hook
-  const [loading] = useState(false); // Replace with DB query hook
+  const { data: databaseResponse, isLoading } =
+    useDocument.document.get(docName);
+  const [data, setData] = useState<JsonData | undefined>();
+  const [jsonSchema, setJsonSchema] = useState<JsonSchema | undefined>();
+  const [uiSchema, setUiSchema] = useState<UISchemaElement | undefined>();
   const [saving, setSaving] = useState(false);
   const [error, setError] = useState<string | false>(false);
+  const [isDirty, setIsDirty] = useState<boolean>();
   const t = useTranslation('common');
   const { success, error: errorNotification } = useNotification();
   const navigate = useNavigate();
-  const { isDirty, setIsDirty } = useDirtyCheck();
+
+  useEffect(() => {
+    if (!databaseResponse) return;
+
+    const {
+      data,
+      // @ts-ignore
+      documentRegistry: { jsonSchema, uiSchema },
```

Comment:
Not intending to leave this here, but needed it for now to compile. Typescript is complaining that:
```
Property 'jsonSchema' does not exist on type '{ __typename: "DocumentRegistryNode"; uiSchemaType: string; documentType: string; context: DocumentRegistryNodeContext; jsonSchema: any; uiSchema: any; } | null | undefined'.ts(2339)
```
But those fields are clearly defined in the auto-generated types. Any ideas?

Reply:
don't know, maybe the linter doesn't like `any` types...